src/testplan/testplanTestRunner.ts

```diff
@@ -0,0 +1,225 @@
+
+        this.logger.log('info', `Running ${testplanPath} as an executable`);
+        return runProcess(
+            config.pythonPath(),
+            [testplanPath].concat(args),
+            {
+                cwd: config.getCwd(),
+                environment: env,
+                acceptedExitCodes: TESTPLAN_NON_ERROR_EXIT_CODES,
+            });
+    }
+
+    private async loadEnvironmentVariables(config: IWorkspaceConfiguration): Promise<IEnvironmentVariables> {
+        const envFileEnvironment = await EnvironmentVariablesLoader.load(config.envFile(), process.env, this.logger);
+
+        const updatedPythonPath = [
+            config.getCwd(),
+            envFileEnvironment.PYTHONPATH,
+            process.env.PYTHONPATH,
+            DISCOVERY_OUTPUT_PLUGIN_INFO.PACKAGE_PATH
+        ].filter(item => item).join(path.delimiter);
+
+        const updatedTestplanPlugins = [
+            envFileEnvironment.TESTPLAN_PLUGINS,
+            DISCOVERY_OUTPUT_PLUGIN_INFO.MODULE_NAME
+        ].filter(item => item).join(',');
+
+        return {
+            ...envFileEnvironment,
+            PYTHONPATH: updatedPythonPath,
+            TESTPLAN_PLUGINS: updatedTestplanPlugins,
+        };
+    }
+
+    private async getJunitReportPath(
+        cwd: string,
+        runArguments: IRunArguments
+    ): Promise<{ dirName: string, cleanupCallback: () => void }> {
+        if (runArguments.junitReportPath) {
+            return Promise.resolve({
+                dirName: path.resolve(cwd, runArguments.junitReportPath),
+                cleanupCallback: () => { /* intentionally empty */ },
+            });
+        }
+        return await this.createTemporaryDirectory();
+    }
+
+    private getDiscoveryArguments(rawTestplanArguments: string[]): string[] {
+        const argumentParser = this.configureCommonArgumentParser();
+        const [, argumentsToPass] = argumentParser.parse_known_args(rawTestplanArguments);
+        return ['--info', 'pattern-full'].concat(argumentsToPass);
+    }
+
+    private getRunArguments(test: string, rawTestplanArguments: string[]): IRunArguments {
+        const argumentParser = this.configureCommonArgumentParser();
+
+        const [knownArguments, argumentsToPass] = argumentParser.parse_known_args(rawTestplanArguments);
+        return {
+            junitReportPath: (knownArguments as { xmlpath?: string }).xmlpath,
+            argumentsToPass: argumentsToPass.concat(
+                test !== this.adapterId ?
+                    ['--pattern', test] :
+                    (knownArguments as { tests?: string[] }).tests || []
+            ),
+        };
+    }
+
+    private configureCommonArgumentParser() {
+        const argumentParser = new ArgumentParser({
+            exit_on_error: false,
+        });
+        argumentParser.add_argument(
+            '--runpath',
+            { action: 'store', dest: 'runpath'});
+        argumentParser.add_argument(
+            '--timeout',
```

Comment:
This list of options for pytest was to filter out options that might break test discovery or execution. All other options (not from this list) will be passed to a runner (testplan in your case). So I think it's safe to remove `timeout`, `shuffle` and `verbose` option from this list.

Reply:
okay